Classes/Domain/Model/Subscription/Subscriber.php

```diff
@@ -0,0 +1,238 @@
+<?php
+declare(strict_types=1);
+
+namespace PhpList\PhpList4\Domain\Model\Subscription;
+
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\HasLifecycleCallbacks;
+use Doctrine\ORM\Mapping\PrePersist;
+use Doctrine\ORM\Mapping\Table;
+use PhpList\PhpList4\Domain\Model\Interfaces\CreationDate;
+use PhpList\PhpList4\Domain\Model\Interfaces\Identity;
+use PhpList\PhpList4\Domain\Model\Interfaces\ModificationDate;
+use PhpList\PhpList4\Domain\Model\Traits\CreationDateTrait;
+use PhpList\PhpList4\Domain\Model\Traits\IdentityTrait;
+use PhpList\PhpList4\Domain\Model\Traits\ModificationDateTrait;
+
+/**
+ * This class represents asubscriber who can subscribe to multiple subscriber lists and can receive email messages from
+ * campaigns for those subscriber lists.
+ *
+ * @Entity(repositoryClass="PhpList\PhpList4\Domain\Repository\Subscription\SubscriberRepository")
+ * @Table(name="phplist_user_user")
+ * @HasLifecycleCallbacks
+ *
+ * @author Oliver Klee <oliver@phplist.com>
+ */
+class Subscriber implements Identity, CreationDate, ModificationDate
+{
+    use IdentityTrait;
+    use CreationDateTrait;
+    use ModificationDateTrait;
+
+    /**
+     * @var \DateTime|null
+     * @Column(type="datetime", nullable=true, name="entered")
+     */
+    protected $creationDate = null;
+
+    /**
+     * @var \DateTime|null
+     * @Column(type="datetime", nullable=true, name="modified")
+     */
+    protected $modificationDate = null;
+
+    /**
+     * @var string
+     * @Column(unique=true)
+     */
+    private $email = '';
+
+    /**
+     * @var bool
+     * @Column(type="boolean")
+     */
+    private $confirmed = false;
+
+    /**
+     * @var bool
+     * @Column(type="boolean")
+     */
+    private $blacklisted = false;
+
+    /**
+     * @var int
+     * @Column(type="integer", name="bouncecount")
+     */
+    private $bounceCount = 0;
+
+    /**
+     * Note: The uniqueness of this column will not be enforced as long as we use the old DB schema,
```

Comment:
Will not be enforced by the database engine, or will not be enforced at all (is there not application-level verification)?

Reply:
It currently will not be enforced at all (as this only influences that generated DB schema).

When we let Doctrine generate the DB schema for us, then this will be enforced (at the DB level).

Reply:
AFAIK, Doctrine only enforces this at the DB level, not at application level.

Reply:
(I havent't checked whether there are any additional checks in phpList 3 for this, though.)

Reply:
Generating duplicate IDs should be impossible using the HMAC system implemented, as I understand it. I think it's enough to rely on phpList3 to validate during generation and DB schema to enforce later when doctrine is used to generate it.

Reply:
No, it's not impossible, just very unlikely that two autogenerated UUID are the same (both in phpList 3 and 4). phpList 3 also does not check for duplicate UUIDs when generating them - it just hopes for the best (fingers crossed).

And it's not HMAC - just a bunch of random bytes.

Reply:
(At least in the UUID-generating piece of code which I found. Maybe there are more than one.)

Reply:
> No, it's not impossible, just very unlikely that two autogenerated UUID are the same (both in phpList 3 and 4). phpList 3 also does not check for duplicate UUIDs when generating them - it just hopes for the best (fingers crossed).
> 
> And it's not HMAC - just a bunch of random bytes.

@michield Please can you clarify the following: does phpList 3 rely on database-level enforcement of UUID uniqueness at the time that the UUIDs are generated?